errors.go

```diff
@@ -146,6 +146,42 @@ func IsBadParameter(e error) bool {
 	return ok
 }
 
+// NotImplemented returns a new instance of NotImplementedError
+func NotImplemented(message string, args ...interface{}) error {
+	return WrapWithMessage(&NotImplementedError{
+		Message: fmt.Sprintf(message, args...),
+	}, message, args...)
+}
+
+// NotImplementedError means that certain functionality hasn't been implemented yet
+type NotImplementedError struct {
+	Message string `json:"message"`
+}
+
+// Error returns log friendly description of an error
+func (e *NotImplementedError) Error() string {
+	return e.Message
+}
+
+// OrigError returns original error
+func (e *NotImplementedError) OrigError() error {
+	return e
+}
+
+// IsNotImplementedError indicates that this error is of NotImplementedError type
+func (e *NotImplementedError) IsNotImplementedError() bool {
+	return true
```

Comment:
If this is degenerate condition - i.e. it is always supposed to return `true` - I'd keep it as not returning a value - otherwise, you'd need to check if that's so in `IsNotImplemented` (as an implementation can return `false`):
```go
func IsNotImplemented(e error) bool {
	type ni interface {
		IsNotImplementedError() bool
	}
	err, ok := Unwrap(e).(ni)
	return ok && err.IsNotImplementedError()
}
```